qir/qsharp-target-tests/TargetTestScripts/Decompositions/SWAP.qs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace QATSample {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Measurement;
+
+    @EntryPoint()
+    operation Main(): Result
+    {
+      use qubit1 = Qubit();
+      use qubit2 = Qubit();
+
+      SWAP(qubit1, qubit2);
```

Comment:
Why do you need a test for both SWAP and SWAPFromSinglyControlled? The difference between the two is purely based on what target package it is loaded with.

Reply:
It is unclear to me what chooses either of the two implementations. If this is only done through the Q# configuration file, then it suffices with a single file that tests otherwise each file should open namespaces with the respective implementations.